core/src/androidTest/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/YouTubePlayerViewClickTest.kt

```diff
@@ -40,7 +41,9 @@ class YouTubePlayerViewClickTest {
     fun testSingleClickListener() {
         // Prepare
         val mockClickListener = mock(View.OnClickListener::class.java)
-        youTubePlayerView.setOnClickListener(mockClickListener)
+        runOnUiThread {
+            youTubePlayerView.setOnClickListener(mockClickListener)
+        }
```

Comment:
Why do we have to do this here and not everywhere else?

Reply:
This is the only failing portion. The others work without it.

Reply:
Do you know what's the reason? It seems odd.

Reply:
Nope, I have absolutely no clue. I even knew the bug existed when I tested on the emulator

Reply:
I think we should understand what's happening. It might be a symptom of a bigger problem.

Reply:
Looked everywhere, but no one else has encountered this issue :(

Should I file a bug on the android issue tracker?

Reply:
mmm if this is a bug on android it should be fairly common. I would say it's on our side. I'll have a look.

Reply:
Looks like threading is a bit strange in instrumentation tests. I don't have time to properly understand it right now. So I'd say we can leave it as it is.

When I run the tests SwipeDown and SwipeRight also fail. Are you aware of it?

Reply:
No, those didn't fail on the emulator and my physical device.

Reply:
So the `setOnLongClickListener()` is the only one that doesn't fail? This is strange

Reply:
Yeah, i had multiple tests failing. Let's submit this one and see what happens once we run tests in travis.